.github/workflows/continous-integration-docker.yml

```
@@ -239,8 +246,10 @@ jobs:
       height_grep_text: 'finished initial sync to chain tip, using gossiped blocks sync_percent=100.* current_height=Height'
 
   # Test that Zebra can answer a synthetic RPC call, using a cached Zebra tip state
+  #
+  # If the state version has changed, waits for the new cached state to be created.
   lightwalletd-rpc-test:
-    needs: build
+    needs: [build, test-full-sync]
```

`lightwalletd-rpc-test` running after a `test-full-sync` will mostly fail as most of the times the full sync is timing out, and then this test will also fail.

Is this a hard dependency or a soft dependency? If this is a soft dependency (it doesn't require to run from the `latest` cached state everytime) then we should make a PR to create a job (`use-cached-state`) to analyze this requirements and depend on that one.


```suggestion
    needs: [build]
```

---

The goal of this change is to implement ticket #4153:
> If a cached state with that version can't be found, the state should be rebuilt.
> (And then the tests should be run with that new state.)

> `lightwalletd-rpc-test` running after a `test-full-sync` will mostly fail as most of the times the full sync is timing out, and then this test will also fail.

Full sync timeouts are a separate bug, I'd like to deal with them in a separate issue.

And they seem to work fine sometimes:
https://github.com/ZcashFoundation/zebra/runs/6401961075?check_suite_focus=true

> Is this a hard dependency or a soft dependency? If this is a soft dependency (it doesn't require to run from the `latest` cached state everytime) then we should make a PR to create a job (`use-cached-state`) to analyze this requirements and depend on that one.

This is a conditional hard dependency.

If the state version has changed, a state rebuild is required, before we run tests that use cached state.
- Before PR #4391, each test does a checkpoint sync or a full sync, because the state version doesn't match (or some tests fail because the new state version is not in the cache)
- After PR #4391, the test fails because a cached state disk with a matching version is not available

If the state version hasn't changed, the state rebuild is skipped, and the tests use a previously cached state.